src/Locator.php

```diff
@@ -199,13 +200,13 @@ protected function search_elements_by_tag($name, &$done, $feeds)
                     ];
 
                     try {
-                        $response = $this->get_http_client()->request(Client::METHOD_GET, $href, $headers);
-                    } catch (HttpException $th) {
-                        continue;
-                    }
+                        $feed = $this->get_http_client()->request(Client::METHOD_GET, $href, $headers);
 
-                    if ((! preg_match('/^http(s)?:\/\//i', $response->get_requested_uri()) || ($response->get_status_code() === 200 || $response->get_status_code() > 206 && $response->get_status_code() < 300)) && $this->is_feed($response, true)) {
-                        $feeds[$href] = $response;
+                        if ((!Misc::is_remote_uri($feed->get_final_requested_uri()) || ($feed->get_status_code() === 200 || $feed->get_status_code() > 206 && $feed->get_status_code() < 300)) && $this->is_feed($feed, true)) {
```

Comment:
We should move this line outside of the `try` block because we don't expect it to throw a `HttpException`.

Reply:
I moved it inside to be skipped when the exception is raised to fix the issue mentioned in https://github.com/simplepie/simplepie/pull/774#discussion_r1141241016.

I guess we could keep it outside of try and duplicate the `$done[] = $href;` assignment in the `catch` clause instead but neither solution feels particularly elegant to me.